modules/app.js

```diff
@@ -0,0 +1,298 @@
+    this.infoClass = appConfigObj.infoClass;
+    this.keysObj = keysObj;
+    this.language = this.getLanguage();
+    this.caps = false;
+  }
+
+  init() { // создать разметку (textarea, клавиатура с клавишами)
+    this.renderKeyboardContainer();
+    this.renderKeyboard(this.keysObj, this.language, this.keyboardKeysClass);
+    this.renderInfo();
+    this.preventDefaultOnTextarea();
+    this.attachEventListeners();
+  }
+
+  attachEventListeners() { // назначить слушатели кликов и нажатий на клавиши
+    this.attachClickEventListeners();
+    this.attachPushEventListeners();
+  }
+
+  attachClickEventListeners() { // назначить слушатели кликов
+    const container = document.querySelector(`.${this.keyboardKeysClass}`);
+    const textarea = document.querySelector(`.${this.keyboardTextareaClass} textarea`);
+
+    container.addEventListener('mousedown', (event) => {
+      // делегируем контейнеру клавиатуры клик по кнопке
+      const button = event.target.closest('.button');
+      if (!button) {
+        return;
+      }
+      this.processEvent(button, 'click');
+    });
+
+    container.addEventListener('mouseup', (event) => {
+      // делегируем контейнеру клавиатуры клик по кнопке
+      const button = event.target.closest('.button');
+      if (!button) {
+        return;
+      }
+      button.classList.remove('pressed');
+      textarea.focus(); // чтобы фокус оставался в поле после клика по кнопкам вне его
+    });
+  }
+
+  attachPushEventListeners() { // назначить слушатели нажатий на клавиши
+    const textarea = document.querySelector(`.${this.keyboardTextareaClass} textarea`);
+    document.addEventListener('keydown', (event) => {
+      if (event.altKey && event.shiftKey) { // переключение языка
+        this.switchKeyboardLanguage();
+      }
+      const button = document.getElementById(event.code);
+      if (!button) {
+        return;
+      }
+      this.processEvent(button, 'click');
+    });
+
+    document.addEventListener('keyup', (event) => {
+      const button = document.getElementById(event.code);
+      if (!button) {
+        return;
+      }
+      button.classList.remove('pressed');
+      textarea.focus(); // вернуть фокус на textarea
+    });
+  }
+
+  processEvent(element, event = 'push') { // обработка события нажатия/клика
+    const cursorPos = this.getCursorPosition();
+    const textarea = document.querySelector(`.${this.keyboardTextareaClass} textarea`);
+    element.classList.add('pressed');
+    if (!element.classList.contains('special')) { // напечатать обычный символ
+      const text = textarea.value;
+      textarea.value = text.substring(0, cursorPos)
+      + ((this.caps) ? element.firstChild.nodeValue.toUpperCase() : element.firstChild.nodeValue)
+      + text.substring(cursorPos);
+      textarea.selectionStart = cursorPos + 1;
+      textarea.selectionEnd = cursorPos + 1;
+    } else if (element.classList.contains('special') && event === 'click') {
```

Comment:
If you use 'else if' you don't need to check again. Check only (event === 'click')